consensus/src/round_manager.rs

```diff
@@ -1360,40 +1380,70 @@ impl RoundManager {
             return Ok(());
```

Comment:
I think we should move the block exist condition here to make it more consistent, it's not obvious to reviewers that we discard the qc if the block doesn't exist

Reply:
Done.